### ISSUE TITLE
Ability to change replica count on dashboard in Runtime

### ISSUE THREAD

**Issue by MadL1me** [dotnet/aspire#4358]
## Overview

To test failover scenarios (for example in this case: https://github.com/dotnet/aspire/issues/4357) first-class support for changing replica count in dashboard would be very helpful 

This feature can be implemented closely with https://github.com/dotnet/aspire/issues/295, because two scenarios are very similar (restart is basically scaling pods to 0 and than back to replica_count) 

For example, we can start with ReplicaCount = 3 for service, but we can change it to "0" and "4", for example. This will insanely help with testing scenarios of failovers locally - test what happens if you just turn off the database, and than rescale it

This would like smth like this:

![image_2024-06-02_00-06-29](https://github.com/dotnet/aspire/assets/46647517/eb97c13b-7225-4277-9883-78e8af90011f)

This will also resolve issue https://github.com/dotnet/aspire/issues/3095 - as one of the ways to perform "complete stop" of the service (or even restart, after some time, with specified replica count we want)

However, It seems easy to add this feature for .NET apps, but it's might be not so trivial for applications where nodes are required to be in state of consensus (any leader-election, multimaster or master-replica architecture) (for example Kafka or Postgres), because it might be required to add info to all brokers/servers about new added host.

**Comment by mitchdenny** [comment]
Yes this is something we definitely want to do. Adding it to the backlog for now.

**Comment by adamint** [comment]
We should create a meta replica set issue, which should include this one.

**Comment by leslierichardson95** [comment]
@karolz-ms Is this suggestion feasible?

**Comment by karolz-ms** [comment]
@leslierichardson95 yes, I do not see the reason why not. The app model has total control over how many replicas there are at any given time.

**Comment by davidfowl** [comment]
But we’re not doing it now as the model is one shot at the moment. We’re slowly getting to a place where it can be more dynamic

**Comment by En3Tho** [comment]
@davidfowl I'm writing a cloud-based load test solution that has a single host instance per test and a variable amount of worker that can change in runtime (host can ask more workers if load testing stretegy implies so). Can this be modelled in Aspire?

**Comment by lukaszblazewicz-ceneo** [comment]
Is there any chance this feature would make it into the 2026 roadmap? Our test suite could greatly benefit from dynamic replica count modification. 
I was also wondering if there’s anything we could implement in the meantime as a workaround? To some extent we were able to achieve our goals with a simple restart scenario via `await app.ResourceCommands.ExecuteCommandAsync(resourceName, KnownResourceCommands.RestartCommand, cancellationToken)` (kudos to [@popytto](https://blog.webrox.fr/how-to-start-or-stop-resources-in-aspire-by-code/)) but I assume that's where the possibility to manage replicas programmatically ends?
